docs/dev-guide/plugin-dev.md

```diff
@@ -34,6 +34,40 @@ So, typical CLI plugin folder structure looks like the following:
 └── ui.js         # Vue UI integration (optional)
 ```
 
+## Naming and discoverability
+
+For a CLI plugin to be usable by other developers, it must be published on [npm](https://www.npmjs.com/) following the name convention `vue-cli-plugin-<name>` or `@scope/vue-cli-plugin-<name>`. Following the name convention allows your plugin to be:
+
+- Discoverable by `@vue/cli-service`;
+- Discoverable by other developers via searching;
+- Installable via `vue add <name>` or `vue invoke <name>`.
+
```

Comment:
Maybe add a warning block here explaining that not following the naming convention will result in problems?

Reply:
Done 😸